core/txbuilder/finalize_test.go

```diff
@@ -33,69 +31,9 @@ import (
 // source, and then building two different txs with that same source,
 // but destinations w/ different addresses.
 func TestConflictingTxsInPool(t *testing.T) {
-	_, db := pgtest.NewDB(t, pgtest.SchemaPath)
-	ctx := context.Background()
-	info, err := bootdb(ctx, db, t)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	_, err = issue(ctx, t, info, info.acctA.ID, 10)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	dumpState(ctx, t, db)
-	prottest.MakeBlock(t, info.Chain)
-	dumpState(ctx, t, db)
-	<-info.pinStore.WaitForPin(account.PinName, info.Chain.Height())
-
-	assetAmount := bc.AssetAmount{
-		AssetID: info.asset.AssetID,
-		Amount:  10,
-	}
-	spendAction := info.NewSpendAction(assetAmount, info.acctA.ID, nil, nil)
-	dest1 := info.NewControlAction(assetAmount, info.acctB.ID, nil)
-
-	// Build the first tx
-	firstTemplate, err := Build(ctx, nil, []Action{spendAction, dest1}, time.Now().Add(time.Minute))
-	if err != nil {
-		testutil.FatalErr(t, err)
-	}
-	coretest.SignTxTemplate(t, ctx, firstTemplate, &info.privKeyAccounts)
-	tx := bc.NewTx(*firstTemplate.Transaction)
-	err = FinalizeTx(ctx, info.Chain, tx)
-	if err != nil {
-		testutil.FatalErr(t, err)
-	}
-
-	// Make the utxo available for reserving again
-	err = cancel(ctx, db, []bc.Outpoint{firstTemplate.Transaction.Inputs[0].TypedInput.(*bc.SpendInput).Outpoint})
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	// Build the second tx
-	dest2 := info.NewControlAction(assetAmount, info.acctB.ID, nil)
-	secondTemplate, err := Build(ctx, nil, []Action{spendAction, dest2}, time.Now().Add(time.Minute))
-	if err != nil {
-		t.Fatal(err)
-	}
-	coretest.SignTxTemplate(t, ctx, secondTemplate, &info.privKeyAccounts)
-	err = FinalizeTx(ctx, info.Chain, bc.NewTx(*secondTemplate.Transaction))
-	if err != nil {
-		testutil.FatalErr(t, err)
-	}
-
-	// Make a block, which should reject one of the txs.
-	dumpState(ctx, t, db)
-	b := prottest.MakeBlock(t, info.Chain)
-	<-info.pinStore.WaitForPin(account.PinName, info.Chain.Height())
-
-	dumpState(ctx, t, db)
-	if len(b.Transactions) != 1 {
-		t.Errorf("got block.Transactions = %#v\n, want exactly one tx", b.Transactions)
-	}
+	// TODO(jackson): Figure out a proper interface to be able
+	// to cancel the reservation.
```

Comment:
There's no reservation ID propagated through the template. Yeah, I think it should be done before landing the PR, but I wanted to get thoughts on how to accomplish it.